packages/kokkos/cmake/kokkos_tpls.cmake

```diff
@@ -2,22 +2,44 @@ KOKKOS_CFG_DEPENDS(TPLS OPTIONS)
 KOKKOS_CFG_DEPENDS(TPLS DEVICES)
 
 FUNCTION(KOKKOS_TPL_OPTION PKG DEFAULT)
+  CMAKE_PARSE_ARGUMENTS(PARSED
+    ""
+    "TRIBITS"
+    ""
+    ${ARGN})
+
+  IF (PARSED_TRIBITS)
+    #this is also a TPL option you can activate with Tribits
+    #Tribits brings its own default
+    SET(DEFAULT TPL_ENABLE_${PARSED_TRIBTS})
+  ENDIF()
+
   KOKKOS_ENABLE_OPTION(${PKG} ${DEFAULT} "Whether to enable the ${PKG} library")
   KOKKOS_OPTION(${PKG}_DIR "" PATH "Location of ${PKG} library")
   SET(KOKKOS_ENABLE_${PKG} ${KOKKOS_ENABLE_${PKG}} PARENT_SCOPE)
   SET(KOKKOS_${PKG}_DIR  ${KOKKOS_${PKG}_DIR} PARENT_SCOPE)
+
+  IF (KOKKOS_HAS_TRILINOS
+      AND KOKKOS_ENABLE_${PKG}
+      AND NOT PARSED_TRIBITS)
+      #this TPL was enabled, but it is not valid to use inside of TriBITS
+      MESSAGE(FATAL_ERROR "Enabled TPL ${PKG} inside TriBITS build, "
+              "but this can only be enabled in a standalone build")
+  ENDIF()
 ENDFUNCTION()
 
 KOKKOS_TPL_OPTION(HWLOC   Off)
 KOKKOS_TPL_OPTION(LIBNUMA Off)
 KOKKOS_TPL_OPTION(MEMKIND Off)
-KOKKOS_TPL_OPTION(CUDA    Off)
+KOKKOS_TPL_OPTION(CUDA    Off TRIBITS CUDA)
```

Comment:
Curious why the appended `TRIBITS CUDA` was needed and what issue it resolved (or prevented)?